accord-core/src/main/java/accord/utils/ReducingIntervalMap.java

```diff
@@ -93,35 +99,47 @@ public boolean equals(Object o)
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         ReducingIntervalMap that = (ReducingIntervalMap) o;
-        return Arrays.equals(ends, that.ends) && Arrays.equals(values, that.values);
+        return Arrays.equals(starts, that.starts) && Arrays.equals(values, that.values);
     }
 
     public int hashCode()
     {
         return Arrays.hashCode(values);
```

Comment:
_Everything_ must have the same inclusive or exclusive starts in the system basically, so it would be a more serious bug if we're ever comparing things where they don't match. But, I don't mind inserting them if you like.

Reply:
Also, hashCode() does not need to include everything, no. It just needs to be the case that equals() => equal hashCode()

Reply:
>In C* we are consistent, but in accord its a library, so a different implementation might mix, so worth detecting

My point is that, whether or not they are mixed, if you provide the wrong value it is a bug. Providing _consistently_ the wrong value is just as much a bug as providing one of them correctly and the other incorrectly.